llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -223,6 +224,8 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
       })
       .clampScalar(0, sXLen, sXLen);
 
+  getActionDefinitionsBuilder(G_IS_FPCLASS).custom();
```

Comment:
Please rebase and then use `customIf(typeIsScalarFPArith(1, ST)` so that we don't do the custom lowering if F or D isn't enabled.

Reply:
Rebased and fixed.